doc/source/data/api/dataset.rst

```diff
@@ -3,289 +3,160 @@
-    :nosignatures:
-
-    ray.data.Dataset.sort
-    ray.data.Dataset.random_shuffle
-    ray.data.Dataset.randomize_block_order
-    ray.data.Dataset.repartition
-
-**Splitting and Merging Datasets**
-
-.. autosummary::
-    :nosignatures:
-
-    ray.data.Dataset.split
-    ray.data.Dataset.split_at_indices
-    ray.data.Dataset.split_proportionately
-    ray.data.Dataset.train_test_split
-    ray.data.Dataset.union
-    ray.data.Dataset.zip
-
-**Grouped and Global Aggregations**
-
-.. autosummary::
-    :nosignatures:
-
-    ray.data.Dataset.groupby
-    ray.data.Dataset.aggregate
-    ray.data.Dataset.sum
-    ray.data.Dataset.min
-    ray.data.Dataset.max
-    ray.data.Dataset.mean
-    ray.data.Dataset.std
-
-**Converting to Pipelines**
-
-.. autosummary::
-    :nosignatures:
-
-    ray.data.Dataset.repeat
-    ray.data.Dataset.window
-
-**Consuming Datasets**
-
-.. autosummary::
-    :nosignatures:
-
-    ray.data.Dataset.show
-    ray.data.Dataset.take
-    ray.data.Dataset.take_all
-    ray.data.Dataset.iter_rows
-    ray.data.Dataset.iter_batches
-    ray.data.Dataset.iter_torch_batches
-    ray.data.Dataset.iter_tf_batches
-
-**I/O and Conversion**
-
-.. autosummary::
-    :nosignatures:
-
-    ray.data.Dataset.write_parquet
-    ray.data.Dataset.write_json
-    ray.data.Dataset.write_csv
-    ray.data.Dataset.write_numpy
-    ray.data.Dataset.write_datasource
-    ray.data.Dataset.to_torch
-    ray.data.Dataset.to_tf
-    ray.data.Dataset.to_dask
-    ray.data.Dataset.to_mars
-    ray.data.Dataset.to_modin
-    ray.data.Dataset.to_spark
-    ray.data.Dataset.to_pandas
-    ray.data.Dataset.to_pandas_refs
-    ray.data.Dataset.to_numpy_refs
-    ray.data.Dataset.to_arrow_refs
-    ray.data.Dataset.to_random_access_dataset
-
-**Inspecting Metadata**
-
-.. autosummary::
-    :nosignatures:
-
-    ray.data.Dataset.count
-    ray.data.Dataset.schema
-    ray.data.Dataset.default_batch_format
-    ray.data.Dataset.num_blocks
-    ray.data.Dataset.size_bytes
-    ray.data.Dataset.input_files
-    ray.data.Dataset.stats
-    ray.data.Dataset.get_internal_block_refs
-
-**Execution**
-
-.. autosummary::
-    :nosignatures:
-
-    ray.data.Dataset.fully_executed
-    ray.data.Dataset.is_fully_executed
-    ray.data.Dataset.lazy
-
-**Serialization**
-
-.. autosummary::
-    :nosignatures:
-
-    ray.data.Dataset.has_serializable_lineage
-    ray.data.Dataset.serialize_lineage
-    ray.data.Dataset.deserialize_lineage
+   Dataset
 
 Basic Transformations
 ---------------------
 
-.. automethod:: ray.data.Dataset.map
-
-.. automethod:: ray.data.Dataset.map_batches
-
-.. automethod:: ray.data.Dataset.flat_map
-
-.. automethod:: ray.data.Dataset.filter
-
-.. automethod:: ray.data.Dataset.add_column
-
-.. automethod:: ray.data.Dataset.drop_columns
-
-.. automethod:: ray.data.Dataset.select_columns
-
-.. automethod:: ray.data.Dataset.random_sample
-
-.. automethod:: ray.data.Dataset.limit
+.. autosummary::
+   :toctree: doc/
+   :nosignatures:
```

Comment:
I feel like not having the function signatures really hurts understanding the one-liner API description, and the [Pandas](https://pandas.pydata.org/docs/reference/groupby.html) and [PySpark](https://spark.apache.org/docs/latest/api/python/reference/pyspark.pandas/groupby.html) docs _do_ include the signatures; can you say more about why you decided to not have the signatures displayed?

Reply:
I guess I am on the other side, that feels the signature is not that useful in one-liner description, given it's also truncated in UI. But I agree this is what other systems are doing. I can add it if this is what people want. WDYT? @jianoaix

Reply:
Got it, I get that perspective as well! My opinion is that, for most of the Datasets APIs, the first argument or two are typically much more important than the remaining arguments, and even with the signature being truncated, the names of the first few arguments adds a lot of valuable context.

For example:
```
read_parquet(paths[, filesystem, parallelism, ...])

map_batches(fn[, batch_size, batch_format, ...])
```

+1 on getting @jianoaix's perspective, we can do a quick 3-person majority rule here, if needed!

Reply:
Yeah, I personally like signatures (and type annotations in general, although may not fit here) and they do convey useful information to the readers (especially when with good namings :).

Reply:
Cool, let me add signature then.